### PR TITLE
Generating typings

### PR DESCRIPTION
**Pull request by demipixel**
Issues/Dilemmas

- [x] Is it okay to generate typings? I feel like this library is built to be more static while minecraft-data updates, but I'm not sure if this would fit in minecraft-data. One way to fix this it to generate the typings on install.
- [x] Right now the test isn't run. I was running it with `tsc test-typings.ts && node test-typings.js; rm test-typings.js;` which definitely isn't ideal :P
- [x] With the library I'm using for json schema -> ts typings, it doesn't seem to like the `protocol_types` folder. More specifically, the lines with `"compareTo":{"$ref": "definitions#/definitions/contextualizedFieldName"}`, it complains that it can't find any "definitions" file in the `protocol_types` folder. Not sure how to fix this one
- [x] There's a few things I added `any` too since I'm not totally sure on their exact format for now.

Otherwise, VSCode seems to recognize the types correctly, and the test throws an error when there is an invalid type passed or whatnot.

### CODE REVIEW

**Comment by demipixel** [comment]
This is probably better put in its own typings module.

**Comment by rom1504** [comment]
no if it should be anywhere, it should be here

**Comment by demipixel** [comment]
Alright, I think it's all done. The two schemas I'm ignoring I don't believe are used by node-minecraft-data, and the only thing using `any` now is `minecraft-data.schemas.*`.

File: doc/api.md [hunk 1]
```diff
@@ -195,27 +195,27 @@ Array of pe supported versions
 
 ## Schemas
 
-### minecraft-data.schema.biomes
```
Comment (rom1504) [inline doc/api.md]: That would require to release a major version. Why do we need to change that ?

Comment (demipixel) [reply doc/api.md]: [The API is wrong](https://github.com/PrismarineJS/node-minecraft-data/blob/8abf8ed0ff9684bd71ecc1a9b980d94540f16549/index.js#L71), I was just fixing it.

**Comment by rom1504** [comment]
merged, thanks !

**Pull request closed without merging**